### ISSUE TITLE
CUDA error in examples/example_bar_pass_prediction.py 

### ISSUE THREAD

**Issue by akiFQC** [MasashiSode/fairtorch#24]
CUDA error occured when I excute  `examples/example_bar_pass_prediction.py `

My environment is here
- ubuntu 18.04
- pyenv 
- GTX1080Ti
- CUDA 11.0
 - #6c5558ad85d79ebf8d3453c737f023e3033c9395

The Command I ran and the following error messege are here.
```
akifqcspc:~/Codes/fairtorch$ python examples/example_bar_pass_prediction.py 
/home/akifukuchi/.pyenv/versions/env_fairtorch/lib/python3.7/site-packages/pandas/core/indexing.py:1743: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  isetter(ilocs[0], value)
Traceback (most recent call last):
  File "examples/example_bar_pass_prediction.py", line 359, in <module>
    metrics_best, train_result_best, valid_result_best = trainer.fit()
  File "examples/example_bar_pass_prediction.py", line 257, in fit
    train_result = self.training_step(epoch)
  File "examples/example_bar_pass_prediction.py", line 179, in training_step
    loss.backward()
  File "/home/akifukuchi/.pyenv/versions/env_fairtorch/lib/python3.7/site-packages/torch/tensor.py", line 221, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/home/akifukuchi/.pyenv/versions/env_fairtorch/lib/python3.7/site-packages/torch/autograd/__init__.py", line 132, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: CUDA error: an illegal memory access was encountered
```

**Comment by MasashiSode** [comment]
does it fix if you merge the latest master into #22 ?

**Comment by MasashiSode** [comment]
#23 fixed this issue.
